test/end-to-end/hup-does-not-abandon-services.exp

```diff
@@ -99,7 +99,7 @@ expect {
         error "Redis not ready to accept connections!"
     }
 }
-set redis_pid [exec pgrep redis-server]
+set redis_pid [exec cat /hab/svc/redis/PID]
```

Comment:
When I run this test locally, it will sometimes fail at this part; it looks like there might be a race between when the Supervisor writes this PID file and when the test tries to read it.

If I add a `sleep 1` immediately before this check, for instance, I see no failures.

Reply:
https://github.com/habitat-sh/habitat/pull/6849/commits/0d5487c41d036bf913827988d68fa58fa4cf3dd1 should address this

Reply:
That works.

Out of curiosity, what prompted you to move away from the `pgrep` approach to the PID file? Not having to worry about whether `pgrep` would be there?

Reply:
As I mentioned in https://github.com/habitat-sh/habitat/pull/6849/commits/c53bfa1eaa8eee7a165a11e501fe82788b384e78:
> there may be other `redis-server`s running locally.

To wit:
```
jbauman@ubuntu:~/habitat$ pgrep -afl redis-server
77398 /opt/opscode/embedded/bin/redis-server 127.0.0.1:16379                         
```